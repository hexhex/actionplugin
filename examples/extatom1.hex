
%
% first component:
% testA returns "foo" if c is empty
% testA returns "bar" if c has one member
%
a(X) :- &testA[c](X), guard(X).
b(X) v c(X) :- a(X).

%
% second component:
% testB returns "bar" if n and m together have 1 or less members
% testB returns "foo" if n and m together have 2 members
%
e(X) :- &testB[m,n](X), guard(X).
m(X) v n(X) :- e(X).
m(foo) :- e(foo).
n(foo) :- e(foo).

%
% third component: collect result from lower components:
%
result(X) :- b(X), m(X).

%
% EDB:
%
guard(foo).
guard(bar).


