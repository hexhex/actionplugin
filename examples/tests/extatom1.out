{n(bar), e(bar), b(foo), guard(bar), guard(foo), a(foo)}
{n(bar), e(bar), guard(bar), guard(foo), c(bar), a(bar)}
{m(bar), e(bar), b(foo), guard(bar), guard(foo), a(foo)}
{m(bar), e(bar), guard(bar), guard(foo), c(bar), a(bar)}
{n(foo), m(foo), e(foo), guard(bar), guard(foo), c(bar), a(bar)}
{result(foo), n(foo), m(foo), e(foo), b(foo), guard(bar), guard(foo), a(foo)}
